packages/block-library/src/cover/style.scss

```diff
@@ -70,6 +70,15 @@
 		opacity: 0.5;
 	}
 
+	@for $i from 1 through 10 {
+		&.has-background-dim.has-background-dim-#{ $i * 10 } {
+			&:not(.has-background-gradient)::before,
+			.wp-block-cover__gradient-background {
+				opacity: $i * 0.1;
+			}
+		}
+	}
+
 	@for $i from 0 through 10 {
 		.wp-block-cover__gradient-background.has-background-dim.has-background-dim-#{ $i * 10 } {
```

Comment:
Just curious: what's the significance of the classname `wp-block-cover__gradient-background`? It's added to the cover blocks' background dim element regardless of whether the overlay color is a gradient or not.

Reply:
As far as I am aware it is only added if there is a gradient background, and so using this rule if that class does not exist the opacity is applied to the ::before of this element, otherwise it is applied to the element itself ... but maybe I will do some additional testing to check that assumption 🤔

Reply:
Wondering if the [span](https://github.com/WordPress/gutenberg/blob/26556937b70bbf4aaef8c112321703b61a740284/packages/block-library/src/cover/edit.js#L726) needs something like `	'wp-block-cover__gradient-background': !! gradientValue,` in the edit and save.js.

Deprecations aside... 😄